src/core/agent/services/signifyNotificationService.ts

```diff
@@ -240,35 +243,38 @@ class SignifyNotificationService extends AgentService {
       opType: OperationPendingRecordType;
     }) => void
   ) {
+    this.pendingOperations = await this.operationPendingStorage.getAll();
     // eslint-disable-next-line no-constant-condition
     while (true) {
-      const pendingOperations = await this.operationPendingStorage.getAll();
-      if (pendingOperations.length > 0) {
-        for (const pendingOperation of pendingOperations) {
+      if (this.pendingOperations.length > 0) {
+        for (const pendingOperation of this.pendingOperations) {
           const operation = await this.signifyClient
             .operations()
             .get(pendingOperation.id);
           if (operation.done) {
             switch (pendingOperation.recordType) {
             case OperationPendingRecordType.Witness: {
-              const aid = await this.identifierStorage.getIdentifierMetadata(
-                pendingOperation.recordId
+              await this.identifierStorage.updateIdentifierMetadata(
+                pendingOperation.recordId,
+                {
+                  isPending: false,
+                }
               );
-              await this.identifierStorage.updateIdentifierMetadata(aid.id, {
-                isPending: false,
-              });
               callback({
                 opType: pendingOperation.recordType,
-                oid: aid.id,
+                oid: pendingOperation.recordId,
```

Comment:
Hmm, is the record ID really just the AID of the identifier (in this case)? Or is it "Witness.<AID>" like on KERIA.

`oid` should be the AID so that the UI just updates based on oid - but we should save it as "Witness.<AID>" in case we have multiple types of long running operations for the same identifier.

Reply:
Will I change oid to recordId?
`
              callback({
                opType: pendingOperation.recordType,
                recordId: pendingOperation.recordId,
`

Reply:
pendingOperation record will be remove after done